core/src/test/java/org/elasticsearch/test/ESSingleNodeTestCase.java

```diff
@@ -118,16 +119,18 @@ protected boolean resetNodeAfterTest() {
 
     private static Node newNode() {
         Node build = NodeBuilder.nodeBuilder().local(true).data(true).settings(Settings.builder()
-            .put(ClusterName.SETTING, InternalTestCluster.clusterName("single-node-cluster", randomLong()))
-            .put("path.home", createTempDir())
-            .put("node.name", nodeName())
-            .put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, 1)
-            .put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, 0)
-            .put("script.inline", "on")
-            .put("script.indexed", "on")
-            .put(EsExecutors.PROCESSORS, 1) // limit the number of threads created
-            .put("http.enabled", false)
-            .put(InternalSettingsPreparer.IGNORE_SYSTEM_PROPERTIES_SETTING, true) // make sure we get what we set :)
+                .put(ClusterName.SETTING, InternalTestCluster.clusterName("single-node-cluster", randomLong()))
+                .put("path.home", createTempDir())
+                .put("path.shared_data", createTempDir().getParent())
```

Comment:
Because the directory set for `path.shared_data` and the custom directories set during tests when indices are created are uncoupled, if I set it to a specific directory in advance, every test (including where we randomly add a custom data path) would be required to know what the data path was already set to.